src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -67,13 +69,17 @@ QtObject:
     result.inAmount = self.currencyService.parseCurrencyValue(result.inSymbol, metadata.amountIn)
     result.outSymbol = item.fromAsset
     result.outAmount = self.currencyService.parseCurrencyValue(result.outSymbol, metadata.amountOut)
+    result.inValue = newCurrencyAmount(result.inAmount, result.inSymbol, 8, true)
```

Comment:
The number of decimal places and whether to remove trailing zeroes depends on the token. The proper currency format needs to be fetched from the currency service (`getCurrencyFormat*(self: Service, symbol: string): CurrencyFormatDto` and `currencyAmountToItem*(amount: float64, format: CurrencyFormatDto) : CurrencyAmount`.

`RootStore.formatCurrencyAmount` can take care of formatting in QML if you pass it the numerical amount and the currency symbol, so perhaps we should stick with the float here instead of building the `CurrencyAmount` object.